src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -765,23 +751,15 @@ private static string RetrieveProcessUserName(Process process)
                 }
 
                 var tokenUser = Marshal.PtrToStructure<Win32Native.TOKEN_USER>(tokenUserInfo);
-
-                // Max username is defined as UNLEN = 256 in lmcons.h
-                // Max domainname is defined as DNLEN = CNLEN = 15 in lmcons.h
-                // The buffer length must be +1, last position is for a null string terminator.
-                int userNameLength = 257;
-                int domainNameLength = 16;
-                Span<char> userNameStr = stackalloc char[userNameLength];
-                Span<char> domainNameStr = stackalloc char[domainNameLength];
-                Win32Native.SID_NAME_USE accountType;
-
-                // userNameLength and domainNameLength will be set to actual lengths.
-                if (!Win32Native.LookupAccountSid(null, tokenUser.User.Sid, userNameStr, ref userNameLength, domainNameStr, ref domainNameLength, out accountType))
+                SecurityIdentifier sid = new SecurityIdentifier(tokenUser.User.Sid);
+                try
                 {
-                    return null;
+                    return sid.Translate(typeof(System.Security.Principal.NTAccount)).Value;
+                }
+                catch (IdentityNotMappedException)
+                {
+                    return sid.Value;
```

Comment:
Is `sid.Value` going to be null when `sid.Translate` fails?

Reply:
`sid.Value` will be the SID as a string, e.g. `S-1-5-...`. I doubt this would ever happen but in case Windows ever adds some new SID type that .NET is unable to translate or someone is doing something weird like crafting their own access tokens with an invalid SID.

Reply:
Why do we want to return the SID as a string in case `Translate` fails? It should be `null` when we cannot get back the user name, just like before your change -- `Win32Native.LookupAccountSid` fails, then return `null`.

Reply:
I just assumed getting something back is better than nothing. This is a very very rare scenario and I would be hard pressed to actually replicate it myself (if it's even at all possible). I can have it return `null` I just didn't really see the benefits.

Reply:
Returning the string SID could cause confusion. User is expecting a name, not a SID. It's better to keep the behavior unchanged in case the SID cannot be mapped to a user name.

Reply:
Updated to return `null`.